### ISSUE TITLE
GET return ETag value not handled by PUT

### ISSUE THREAD

**Issue by petterreinholdtsen** [HiOA-ABI/nikita-noark5-core#77]
The etag content of a GET request look like this: ETag: "0".  But the quoted value is not accepted by PUT.  I suspect the intention was that it should look liek this: ETag: 0.  Anyway, the etag header value from GET should be something that can be passed on as an etag header value to PUT without modifications. See also issue #76 for what happen when a bogus etag value is passed to PUT.

**Comment by petterreinholdtsen** [comment]
See also issue #69.

**Comment by tsodring** [comment]
This is related to #76 and hopefully solved there. 

* Extra quotes are being added when an ETAG is generated server side.  We discovered that the etag is being converted to JSON and the etag value becomes quoted
Fixed with:
https://github.com/HiOA-ABI/nikita-noark5-core/commit/3b1feac41ed8607267ecfd12983267671f15e069

**Comment by tsodring** [comment]
The previous comment is not correct. The ETAG contents are not being converted to JSON and being quoted. What's happening is that spring is adding the quotes automatically.

```
public BodyBuilder eTag(String eTag) {
			if (eTag != null) {
				if (!eTag.startsWith("\"") && !eTag.startsWith("W/\"")) {
					eTag = "\"" + eTag;
				}
				if (!eTag.endsWith("\"")) {
					eTag = eTag + "\"";
				}
			}
			this.headers.setETag(eTag);
			return this;
		}
```
The HTTP spec does not disallow the quotes from being added, nor does it require it. 

  https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.19

This causes an issue for us as we are using numerical values in the ETAG contents, but the addition of the quotes means we have to remove the quotes when the client sends the ETAG back.

The solution is use 

  .header(ETAG, updatedFonds.getVersion().toString())

rather than

  .eTag(updatedFonds.getVersion().toString())

**Comment by tsodring** [comment]
The previous comment is also incorrect ... The quotes are required. Of course spring is doing the right thing!  The client must ensure the quotes are there. The client will probably just give the ETAG header back exactly the way it got it.

Hopefully  https://github.com/HiOA-ABI/nikita-noark5-core/commit/fe9ab963350d82a9eb9647d1b65bbc47a878c2b5 will finally fix this issue.

This update checks that the incoming eTag has quotes and is a numeric value
as what nikita expects. After it has been processed, spring seems to be removing the quotes of the
ETAG so when we retrieve a value in the controller it is without quotes.

**Comment by petterreinholdtsen** [comment]
This issue is fixed now in commit f30b6539de1ba24cbe7c8a8c4f81d3b07b13efce .
